$OpenBSD: patch-Portroach_SiteHandler_CPAN_pm,v 1.1 2014/09/05 15:52:16 jasper Exp $

Ensure we only add a download url for actual hits on CPAN

--- Portroach/SiteHandler/CPAN.pm.orig	Fri Sep  5 15:56:15 2014
+++ Portroach/SiteHandler/CPAN.pm	Fri Sep  5 17:51:22 2014
@@ -115,7 +115,7 @@ sub GetFiles
 
 	    my @hits = @{$json->{hits}->{hits}};
 
-	    push @$files, @hits[0]->{fields}->{download_url};
+	    push(@$files, @hits[0]->{fields}->{download_url}) if (@hits[0]);
 	} else {
 	    my $code = $resp->code;
 	    _debug("GET failed: $code");
